projects/packages/jetpack-mu-wpcom/src/features/wpcom-themes/js/theme-actions.js

```diff
@@ -0,0 +1,32 @@
+const wpcomThemesRemoveWpcomActions = () => {
+	const themeOverlay = document.querySelector( '.theme-overlay' );
+	if ( ! themeOverlay ) {
+		return;
+	}
+
+	const observer = new MutationObserver( mutations => {
+		for ( const mutation of mutations ) {
+			for ( const node of mutation.addedNodes ) {
+				// If this is not an overlay for the active theme, bail and check the next node.
+				if (
+					! node.classList.contains( 'theme-overlay' ) ||
+					! node.classList.contains( 'active' )
+				) {
+					continue;
+				}
+
+				const themeActions = node.querySelector( '.theme-actions .active-theme' );
+				for ( const action of themeActions?.children ?? [] ) {
+					if ( action.getAttribute( 'href' )?.includes( 'https://wordpress.com' ) ) {
+						themeActions.removeChild( action );
+					}
+				}
+				return;
```

Comment:
Should we traverse all mutations instead of returning early?

Reply:
No, we're returning only if the current mutation is the one adding a `.theme-overlay.active` node which is the one for the active theme (and should be only one). We're checking the rest of the mutations.